src/main/java/mandel/citibike/map/MapFrame.java

```diff
@@ -0,0 +1,88 @@
+package mandel.citibike.map;
+
+import javax.swing.*;
+import java.awt.event.*;
+import java.awt.*;
+
+import org.jxmapviewer.*;
+import org.jxmapviewer.painter.CompoundPainter;
+import org.jxmapviewer.painter.Painter;
+import org.jxmapviewer.viewer.*;
+
+import java.util.List;
+
+public class MapFrame extends JFrame {
+    private final MapController controller;
+    private final JTextField from;
+    private final JTextField to;
+
+    public MapFrame() {
+        setTitle("CitiBike Map");
+        setSize(850, 700);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+        RoutePainter routePainter = new RoutePainter();
+        WaypointPainter<Waypoint> waypointPainter = new WaypointPainter<>();
+        JXMapViewer mapViewer = new JXMapViewer();
+
+        MapComponent component = new MapComponent(mapViewer);
+        add(component);
```

Comment:
You're adding the component, which later on is getting overwritten when you add the mapViewer